Backends/CLX/mirror.lisp

```diff
@@ -122,6 +122,21 @@
     (xlib:display-force-output (clx-port-display port))))
 
 (defmethod port-shrink-sheet ((port clx-basic-port) (sheet mirrored-sheet-mixin))
-  (when-let ((mirror (sheet-direct-mirror sheet)))
-    (xlib:iconify-window (window mirror) (clx-port-screen port))
-    (xlib:display-force-output (clx-port-display port))))
+  ;; Not all x11 window-managers iconify windows (in particular many
+  ;; tiling window-managers). In those window managers, after the call
+  ;; of shrink-frame McClim think that the frame is iconified but it
+  ;; is not. -- admich 2021-08-21
+  (when-let ((window (window (sheet-direct-mirror sheet))))
+    (cond
+      ((or (eq 0 (car (xlib:get-property window :WM_STATE)))
+           (not (xlib:get-property window :WM_STATE)))
+       (let ((hints (xlib:wm-hints window)))
+          (setf (xlib:wm-hints-initial-state hints) :iconic)
+          (setf (xlib:wm-hints window) hints)
+          (xlib:map-window window)
+          (xlib:display-force-output (clx-port-display port))))
+      ((eq 1 (car (xlib:get-property window :WM_STATE)))
+       (xlib:iconify-window window (clx-port-screen port))
+       (xlib:display-force-output (clx-port-display port)))
+      ((eq 2 (car (xlib:get-property window :WM_STATE)))
+       (warn "The sheet is already shrunk")))))
```

Comment:
please access the property only once by using let:
```
(let ((wm-state (car (xlib:get-property window :wm_state)))) …)
```

Reply:
also, do we want to signal a warning here? perhaps leaving this as a no-op without further notification would be less engaging for the programmer without much demerit.

Reply:
done --also 2 was the wrong constant, anyway now there is +icccm-iconic-state+ constant